.dotnet.azure/sdk/openai/Azure.AI.OpenAI/src/Custom/AzureOpenAIClientOptions.cs

```diff
@@ -65,12 +64,10 @@ public AzureOpenAIClientOptions(ServiceVersion version = LatestVersion)
     /// <summary> The version of the service to use. </summary>
     public enum ServiceVersion
     {
-        /// <summary> Service version "2024-04-01-preview". </summary>
-        V2024_04_01_Preview = 7,
-        V2024_05_01_Preview = 8,
+#if !AZURE_OPENAI_GA
+        V2024_08_01_Preview = 8,
+#endif
         V2024_06_01 = 9,
```

Comment:
nit: V2024_06_01 should probably start at 0, and then the V2024_08_01_Preview should be 1.